iceoryx2-ffi/ffi/src/api/config.rs

```diff
@@ -12,22 +12,1639 @@
+#[no_mangle]
+pub unsafe extern "C" fn iox2_config_defaults_publish_subscribe_publisher_max_loaned_samples(
+    handle: iox2_config_ref_h,
+) -> c_size_t {
+    debug_assert!(!handle.is_null());
+
+    let config = &*handle.as_type();
+    config
+        .value
+        .as_ref()
+        .value
+        .defaults
+        .publish_subscribe
+        .publisher_max_loaned_samples
+}
+
+/// Sets the maximum amount of [`iox2_sample_mut_h`](crate::api::iox2_sample_mut_h)s a
+/// [`iox2_publisher_h`](crate::api::iox2_publisher_h) can loan at the same time.
+///
+/// # Safety
+///
+/// * `handle` - A valid non-owning [`iox2_config_ref_h`] obtained by [`iox2_cast_config_ref_h`].
+#[no_mangle]
+pub unsafe extern "C" fn iox2_config_defaults_publish_subscribe_set_publisher_max_loaned_samples(
+    handle: iox2_config_ref_h,
+    value: c_size_t,
+) {
+    debug_assert!(!handle.is_null());
+
+    let config = &mut *handle.as_type();
+    config
+        .value
+        .as_mut()
+        .value
+        .defaults
+        .publish_subscribe
+        .publisher_max_loaned_samples = value;
 }
 
+/// Returns the maximum history size a [`iox2_subscriber_h`](crate::api::iox2_subscriber_h) can
+/// request from a [`iox2_publisher_h`](crate::api::iox2_publisher_h).
+///
+/// # Safety
+///
+/// * `handle` - A valid non-owning [`iox2_config_ref_h`] obtained by [`iox2_cast_config_ref_h`].
+#[no_mangle]
+pub unsafe extern "C" fn iox2_config_defaults_publish_subscribe_publisher_history_size(
+    handle: iox2_config_ref_h,
+) -> c_size_t {
+    debug_assert!(!handle.is_null());
+
+    let config = &*handle.as_type();
+    config
+        .value
+        .as_ref()
+        .value
+        .defaults
+        .publish_subscribe
+        .publisher_history_size
+}
+
+/// Sets the maximum history size a [`iox2_subscriber_h`](crate::api::iox2_subscriber_h) can
+/// request from a [`iox2_publisher_h`](crate::api::iox2_publisher_h).
+///
+/// # Safety
+///
+/// * `handle` - A valid non-owning [`iox2_config_ref_h`] obtained by [`iox2_cast_config_ref_h`].
+#[no_mangle]
+pub unsafe extern "C" fn iox2_config_defaults_publish_subscribe_set_publisher_history_size(
+    handle: iox2_config_ref_h,
+    value: c_size_t,
+) {
+    debug_assert!(!handle.is_null());
+
+    let config = &mut *handle.as_type();
+    config
+        .value
+        .as_mut()
+        .value
+        .defaults
+        .publish_subscribe
+        .publisher_history_size = value;
+}
+
+/// Defines how the [`iox2_subscriber_h`](crate::api::iox2_subscriber_h) buffer behaves when it is
+/// full. When safe overflow is activated, the [`iox2_publisher_h`](crate::api::iox2_publisher_h) will
+/// replace the oldest [`iox2_sample_h`](crate::api::iox2_sample_h) with the newest one.
+///
+/// # Safety
+///
+/// * `handle` - A valid non-owning [`iox2_config_ref_h`] obtained by [`iox2_cast_config_ref_h`].
+#[no_mangle]
+pub unsafe extern "C" fn iox2_config_defaults_publish_subscribe_enable_safe_overflow(
```

Comment:
Potentially something for later. I recall that we wanted to change the name or do this at a different place.

I guess the C and C++ bindings will be announced as experimental/beta and this could be changed later on?

Reply:
Yes and yes. @sempuki suggested to get rid of `safe` and you suggested something like circular buffer.

Also the safe overflow property should be part of the subscriber and not the service. There already exists and issue #17 .

If I find time I will try to integrate this into iceoryx2 before the v0.4 otherwise it will be part of v0.5